### PR TITLE
Added config and key generation for oauth signing key

### PR DESCRIPTION
**Pull request by woutslakhorst**
part of #101

### CODE REVIEW

**Comment by woutslakhorst** [comment]
Moving PemToSigner to crypto module

**Comment by woutslakhorst** [comment]
waiting for nuts-foundation/nuts-crypto#98

File: README_options.rst [hunk 1]
```diff
@@ -7,6 +7,8 @@ enableCORS                 false           Set if you want to allow CORS request
 irmaConfigPath                             path to IRMA config folder. If not set, a tmp folder is created.                                                         
 irmaSchemeManager          pbdf            The IRMA schemeManager to use for attributes. Can be either 'pbdf' or 'irma-demo', default: pbdf                         
 mode                                       server or client, when client it does not start any services so that CLI commands can be used.                           
+oAuthKeyGeneration         true            Auto generate OAuth JWT signing key if missing.
```
Comment (reinkrul) [inline README_options.rst:10]: is this casing right? (oAuth/OAuth)

File: engine/engine.go [hunk 2]
```diff
@@ -151,6 +152,8 @@ func flagSet() *pflag.FlagSet {
 	flags.String(pkg.ConfActingPartyCN, defs.ActingPartyCn, "The acting party Common name used in contracts")
 	flags.Bool(irma.ConfSkipAutoUpdateIrmaSchemas, defs.SkipAutoUpdateIrmaSchemas, "set if you want to skip the auto download of the irma schemas every 60 minutes.")
 	flags.Bool(pkg.ConfEnableCORS, defs.EnableCORS, "Set if you want to allow CORS requests. This is useful when you want browsers to directly communicate with the nuts node.")
+	flags.Bool(oauth.ConfGenerateOAuthKeys, defs.GenerateOAuthKeys, "Auto generate OAuth JWT signing key if missing.")
+	flags.String(oauth.ConfOAuthSigningKey, defs.OAuthSigningKey, "Path to PEM encoded private key.")
```
Comment (reinkrul) [inline engine/engine.go:156]: I had to read the code to realise the authorization server isn't enabled when this is empty, maybe add that to the config option explanation.

File: pkg/services/oauth/oauth.go [hunk 3]
```diff
@@ -1,26 +1,95 @@
...
+		fmt.Errorf("failed to parse OAuthSigningKey: %w", err)
+		return
+	}
+
+	s.signer, err = cert.PemToSigner(bytes)
+	return
+}
+
+func (s *OAuthService) generateOAuthSigningKey() (bytes []byte, err error) {
+	var (
```
Comment (reinkrul) [inline pkg/services/oauth/oauth.go:67]: shouldn't this be done by `nuts-crypto` so we can switch to other backends (like vault) more easily in the future?

File: pkg/services/oauth/oauth.go [hunk 4]
```diff
@@ -1,26 +1,95 @@
...
+
+	if err = ioutil.WriteFile(s.OAuthSigningKey, bytes, 0660); err != nil {
+		return
+	}
+
+	pub := ec.Public()
+	if der, err = x509.MarshalPKIXPublicKey(pub); err != nil {
+		return
+	}
+	pubBytes := pem.EncodeToMemory(&pem.Block{Type: "PUBLIC KEY", Bytes: der})
```
Comment (reinkrul) [inline pkg/services/oauth/oauth.go:84]: `EC PRIVATE KEY` vs `PUBLIC KEY`, shouldn't we mention `EC` here as well?

**Review by reinkrul — CHANGES_REQUESTED** [review]
see comments

**Comment by woutslakhorst** [comment]
Going to change it to an always-on service with keys in crypto-storage. Will add some commands for extracting the pub key and key rotation

**Comment by woutslakhorst** [comment]
continued in #111

**Pull request closed without merging**